### ISSUE TITLE
The text-to-image tasks

### ISSUE THREAD

**Issue by 2862909626** [lucidrains/gigagan-pytorch#49]
How do I implement text-to-image training and generation on the LAION dataset?

**Comment by darshanmakwana412** [comment]
You can create a dataloader that passes an (image, text) pair for training, an example of the [MockTextImageDataset](https://github.com/lucidrains/gigagan-pytorch/blob/55b621ee6856d68bd8f99923552f55da53a79888/gigagan_pytorch/data.py#L94C11-L94C27) is also given in the data.py file

**Comment by linjie99i** [comment]
> How do I implement text-to-image training and generation on the LAION dataset?

May I ask, have you tried training the text generation image model now?

**Comment by kaijie47** [comment]
> > How do I implement text-to-image training and generation on the LAION dataset?
> 
> May I ask, have you tried training the text generation image model now?

I tried it out. After adding the text data, the model seemed very unstable. Nan or inf would occur after about 20 training steps.

**Comment by kaijie47** [comment]
This first appears in MAL. The model crashes after about  400 steps whenMAL's weight is set to 0. I have tried to look at the gradient norm and it seems to be at the to_logits layers that the problem occurs first.
